File: app/src/main/java/com/iug/coursehub/feature/user/UserNavHostFragment.java
package com.iug.coursehub.feature.user;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.fragment.app.Fragment;
import androidx.lifecycle.ViewModelProvider;
import androidx.navigation.fragment.NavHostFragment;
import androidx.navigation.ui.NavigationUI;

import com.iug.coursehub.MainActivity;
import com.iug.coursehub.R;
import com.iug.coursehub.databinding.FragmentUserNavHostBinding;

public class UserNavHostFragment extends Fragment {

    private UserNavHostViewModel viewModel;

    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
        viewModel = new ViewModelProvider(this).get(UserNavHostViewModel.class);

        FragmentUserNavHostBinding binding = FragmentUserNavHostBinding.inflate(inflater, container, false);

        NavHostFragment navHostFragment = (NavHostFragment) getChildFragmentManager().findFragmentById(R.id.user_nav_host);
        if (navHostFragment != null) {
            NavigationUI.setupWithNavController(binding.bottomNavigation, navHostFragment.getNavController());
        }

        viewModel.getUiAction().observe(getViewLifecycleOwner(), action -> {
            if (action instanceof UserNavHostViewModel.UiAction.NavigateFromUserToAuth) {
                ((MainActivity) requireActivity()).viewModel.navigateUserToAuth();
            }
        });

        return binding.getRoot();
    }
}

File: app/src/main/java/com/iug/coursehub/feature/user/UserNavHostFragment.kt
package com.iug.coursehub.feature.user

import android.os.Bundle
import androidx.fragment.app.Fragment
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.viewModels
import androidx.navigation.fragment.NavHostFragment
import androidx.navigation.fragment.findNavController
import androidx.navigation.ui.setupWithNavController
import com.iug.coursehub.MainActivity
import com.iug.coursehub.R
import com.iug.coursehub.databinding.FragmentUserNavHostBinding

class UserNavHostFragment : Fragment() {

    override fun onCreateView(
        inflater: LayoutInflater, container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View {
        viewModel = viewModels<UserNavHostViewModel>().value

        val binding = FragmentUserNavHostBinding.inflate(inflater, container, false)

        val navHostFragment = childFragmentManager.findFragmentById(R.id.user_nav_host) as NavHostFragment
        val navController = navHostFragment.navController

        binding.bottomNavigation.setupWithNavController(navController)

        viewModel.uiAction.observe(viewLifecycleOwner) { action ->
            when(action) {
                UserNavHostViewModel.UiAction.NavigateFromUserToAuth -> {
                    (requireActivity() as MainActivity).viewModel.navigateUserToAuth()
                }
            }
        }

        return binding.root
    }

    companion object {
        lateinit var viewModel: UserNavHostViewModel
    }
}